rust/lance-core/src/cache.rs

```diff
@@ -150,8 +151,14 @@ impl LanceCache {
 
     pub fn insert<T: DeepSizeOf + Send + Sync + 'static>(&self, key: &str, metadata: Arc<T>) {
         let key = self.get_key(key);
-        self.cache
-            .insert((key, TypeId::of::<T>()), SizedRecord::new(metadata));
+        let record = SizedRecord::new(metadata);
+        tracing::trace!(
+            target: "lance_cache::insert",
+            key = key,
+            type_id = std::any::type_name::<T>(),
```

Comment:
Is this likely to be useful?

Reply:
I'm using it while testing the cache so I can understand what the final keys are. It's helping me find places where we are duplicating prefixes. I'd like to keep it.